src/vm/threadstatics.h

```diff
@@ -432,6 +439,7 @@ struct ThreadLocalModule
     PTR_DynamicClassInfo     m_pDynamicClassTable;   // used for generics and reflection.emit in memory
     SIZE_T                   m_aDynamicEntries;      // number of entries in dynamic table
     OBJECTHANDLE             m_pGCStatics;           // Handle to GC statics of the module
+    LoaderAllocator*         m_pLoaderAllocator;
```

Comment:
Should the pointer to the LoaderAllocator rather be in in the `DynamicClassInfo`? We are going to have the same pointer on every thread - so we can have it in the global structure that is not per-thread.

Reply:
The `DynamicClassInfo` is per thread. In fact, the `ThreadLocalModule` contains an array of `DynamicClassInfo` entries. If we really have issue with the OFFSETOF__ThreadLocalModule__m_pDataBlob as you've mentioned in the other comment, then it seems that we will need to place the loader allocator into the CollectibleDynamicEntry and live with the fact that we consume a bit more memory.